packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -6,10 +6,26 @@
  */
 import Cookies from 'js-cookie'
 
-export interface BaseStorage {
-    set(key: string, value: string, options?: unknown): void
-    get(key: string): string
-    delete(key: string): void
+export type StorageTypes = 'cookie' | 'local'
+
+export interface BaseStorageOptions {
+    keyPrefix?: string
+    keyPrefixSeperator?: string
+}
+
+export abstract class BaseStorage {
+    options?: BaseStorageOptions
+
+    constructor(options?: BaseStorageOptions) {
+        this.options = options
+    }
+    protected getPrefixedKey(key: string): string {
+        const {keyPrefix, keyPrefixSeperator = '_'} = this?.options || {}
+        return `${keyPrefix ? keyPrefix + keyPrefixSeperator : ''}${key}`
+    }
```

Comment:
```suggestion
    protected options: Required<BaseStorageOptions>

    constructor(options: BaseStorageOptions = {keyPrefixSeparator: '_'}) {
        this.options = {
          keyPrefixSeparator: options.keyPrefix ? options.keyPrefixSeparator ?? '_' : ''
          keyPrefix: options.keyPrefix ?? ''
        }
    }

    protected getPrefixedKey(key: string): string {
        return `${this.keyPrefix}${this.keyPrefixSeparator}${key}`
    }
```
1. Makes life simpler to just do it once, rather than every call. 😉 
2. Typo - it's sep**a**rator
3. Default param isn't necessary, since we need the fallback in the code anyway, but it could be helpful for autocomplete / generated docs.